klab/public/tests.py

```diff
@@ -54,15 +57,27 @@ def setUp(self):
                                               created_by=self.user,
                                               modified_by=self.user)
 
-    def test_main_page_with_post_past_event(self):
+        self.data = ElementTree.fromstring('<photos page="2" pages="89" perpage="10" total="881"><photo id="2636" owner="47058503995@N01" secret="a123456" server="2" title="test_04" ispublic="1" isfriend="0" isfamily="0" /> <photo id="2635" owner="47058503995@N01" secret="b123456" server="2" title="test_03" ispublic="0" isfriend="1" isfamily="1" /><photo id="2633" owner="47058503995@N01" secret="c123456" server="2" title="test_01" ispublic="1" isfriend="0" isfamily="0" /><photo id="2610" owner="12037949754@N01" secret="d123456" server="2" title="00_tall" ispublic="1" isfriend="0" isfamily="0" /></photos>')
+
+        self.mock_response = mock.Mock()
+
+    @mock.patch('flickrapi.FlickrAPI.walk')
+    def test_main_page_with_post_past_event(self,mock_flickr):
         self.create_event(-1, "$500k Investment Lessons From Seedstars World 2016", "This Thursday meet Louis Antoine Muhire, CEO& Founder of Mergims, who is going to clear;y highlight and broadly describe the $500k Investment lessons From Seedstars World 2016.")
+        self.mock_response.etree.return_value = self.data
+        self.mock_response.status_code = 200
+        mock_flickr.return_value = self.mock_response.etree.return_value
```

Comment:
Yes we can, but let say if we want to change later the mock data or flickr's team update it api i.e how the etree is structured, we'll be oblige to change each line we have the return value, right now we've only 3 mocks but for the future  it'll be easy for us to make one update instead of 3 updates(now ) and may be n updates (later). All approach can lead to the same result.